client/src/bundles/ConfirmFolder.js

```diff
@@ -0,0 +1,148 @@
+/* global window */
+import i18n from 'i18n';
+import jQuery from 'jquery';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { loadComponent } from 'lib/Injector';
+
+const FormBuilderModal = loadComponent('FormBuilderModal');
+
+jQuery.entwine('ss', ($) => {
+  /** Mark newly added fields as new */
+  $('#Form_EditForm_Fields').entwine({
+    onmatch() {
+      this._super();
+
+      this.on('addnewinline', () => {
+        this.one('reload', () => {
+          const newField = this.find('.ss-gridfield-item').last();
+          newField.attr('data-new', 1);
+        });
+      });
+    }
+  });
+
+  /** Move our options under the radio button areas */
+  $('#Form_Form_FolderOptions-new').entwine({
+    onmatch() {
+      $('#Form_Form_CreateFolder_Holder').detach().appendTo($('#Form_Form_FolderOptions-new').parent().parent());
+
+      $('#Form_Form_FolderID_Holder').css('display', $(this).prop('checked') ? 'none' : 'none');
+      $('#Form_Form_CreateFolder_Holder').css('display', $(this).prop('checked') ? 'block' : 'none');
+    },
+    onchange() {
+      $('#Form_Form_FolderID_Holder').css('display', $(this).prop('checked') ? 'none' : 'none');
+      $('#Form_Form_CreateFolder_Holder').css('display', $(this).prop('checked') ? 'block' : 'none');
+    }
+  });
+
+  /** Move our options under the radio button areas */
+  $('#Form_Form_FolderOptions-existing').entwine({
+    onmatch() {
+      $('#Form_Form_FolderID_Holder').detach().appendTo($('#Form_Form_FolderOptions-existing').parent().parent());
+
+      $('#Form_Form_FolderID_Holder').css('display', $(this).prop('checked') ? 'block' : 'none');
+      $('#Form_Form_CreateFolder_Holder').css('display', $(this).prop('checked') ? 'none' : 'block');
+    },
+    onchange() {
+      $('#Form_Form_FolderID_Holder').css('display', $(this).prop('checked') ? 'block' : 'none');
+      $('#Form_Form_CreateFolder_Holder').css('display', $(this).prop('checked') ? 'none' : 'block');
+    }
+  });
+
+  /** make the Create Folder text field an input group and prepend with SecureUploads/ */
+  $('#Form_Form_CreateFolder').entwine({
+    onmatch() {
+      $(this).parent().addClass('input-group');
+
+      const inputGroupPrepend = $(
+        '<div class="input-group-prepend">' +
+          '<span class="input-group-text">SecureUploads/</span>' +
+        '</div>'
+      );
+      inputGroupPrepend.insertBefore($(this));
+      $('<br/>').insertAfter($(this));
+    }
+  });
+
+  /**
+   * Kick off a folder confirmation dialog when adding an EditableFileField
+   */
+  $('.uf-field-editor .ss-gridfield-items .dropdown.editable-column-field.form-group--no-label').entwine({
+    onchange() {
+      // if EditableFileField is selected and the field is new
+      if (
+        this.get(0).value === 'SilverStripe\\UserForms\\Model\\EditableFormField\\EditableFileField'
+        && $(this).closest('tr').data('new') === 1
+      ) {
```

Comment:
I wonder if this could target elements more strategically (and fire less often) by shifting [data-new] into the base selector, something like this:

```suggestion
  // Monitor new fields to intercept when EditableFileField is selected
  $('.uf-field-editor .ss-gridfield-items [data-new=1] .dropdown.editable-column-field.form-group--no-label').entwine({
    onchange() {
      // if EditableFileField is selected
      if (
        this.get(0).value === 'SilverStripe\\UserForms\\Model\\EditableFormField\\EditableFileField'
      ) {
```